astropy/utils/introspection.py

```diff
@@ -154,6 +155,12 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
         from pkg_resources import parse_version
     except ImportError:
         from distutils.version import LooseVersion as parse_version
+        # LooseVersion raises a TypeError when strings like dev, rc1 are part
+        # of the version number. Match the dotted numbers only.
+        expr = '^([1-9]\\d*!)?(0|[1-9]\\d*)(\\.(0|[1-9]\\d*))*'
```

Comment:
this seems overly complicated, e.g why not have ``[0-9]`` instead of ``0|[1-9]`` or ``\d``?